File: server/apps/api-service/src/main/java/com/sharetimer/apiservice/application/port/in/command/UpdateTimerCommand.java
package com.sharetimer.apiservice.application.port.in.command;

import java.time.Instant;
import java.time.temporal.ChronoUnit;
import com.sharetimer.web.support.exception.BadRequestException;

public record UpdateTimerCommand(String timerId, String ownerToken, Instant requestTime,
    Instant targetTime) {

  public UpdateTimerCommand {
    if (targetTime.isBefore(Instant.now().plus(1, ChronoUnit.MINUTES))) {
      throw new BadRequestException("InvalidTargetTime",
          "'targetTime' must be at least 1 minute after current time.");
    }
  }

}

File: server/apps/api-service/src/main/kotlin/com/sharetimer/apiservice/application/port/input/command/UpdateTimerCommand.kt
package com.sharetimer.apiservice.application.port.input.command

import com.sharetimer.web.support.exception.BadRequestException
import java.time.Instant
import java.time.temporal.ChronoUnit

data class UpdateTimerCommand(
    val timerId: String,
    val ownerToken: String,
    val requestTime: Instant,
    val targetTime: Instant,
) {
    init {
        if (targetTime.isBefore(Instant.now().plus(1, ChronoUnit.MINUTES))) {
            throw BadRequestException(
                "InvalidTargetTime",
                "'targetTime' must be at least 1 minute after current time.",
            )
        }
    }
}
